### ISSUE TITLE
sails.lower does not close mysql properly

### ISSUE THREAD

**Issue by arkoak** [balderdashy/sails#2309]
Unable to reuse mysql connection after sails is lowered

to reproduce run 
```bash
mocha testfile.js --watch
```
testfile contents below : 
```javascript
before(function (done) {
    var SailsApp = require('sails').Sails;
    var Sails = new SailsApp();
    Sails.lift({
    }, function (err, sails) {
        done(err, sails);
    });
});
describe.skip("dummy test", function () {
        it('is a pending dummy test case', function () {
        });
    });
after(function (done) { 
    sails.lower();
    done();
});
```

When mocha runs the **before** hook 2nd time or more, following error is observed

```
Possibly unhandled AdapterError: Invalid connection name specified
  at getConnectionObject (C:\Users\<user>\explore\nodes\powerapp\node_modules\sails-mysql\lib\adapter.js:1171:35)
  at spawnConnection (C:\Users\<user>\explore\nodes\powerapp\node_modules\sails-mysql\lib\adapter.js:1086:7)
  at Object.module.exports.adapter.find (C:\Users\<user>\explore\nodes\powerapp\node_modules\sails-mysql\lib\adapter.js:790:16)
  at [object Object].module.exports.find (C:\Users\<user>\AppData\Roaming\npm\node_modules\sails\node_modules\waterline\lib\waterline\adapt
er\dql.js:121:13)
  at [object Object].module.exports.findOne (C:\Users\<user>\AppData\Roaming\npm\node_modules\sails\node_modules\waterline\lib\waterline\ad
apter\dql.js:161:10)
  at [object Object]._runOperation (C:\Users\<user>\AppData\Roaming\npm\node_modules\sails\node_modules\waterline\lib\waterline\query\finde
rs\operations.js:396:29)
  at [object Object].run (C:\Users\<user>\AppData\Roaming\npm\node_modules\sails\node_modules\waterline\lib\waterline\query\finders\operati
ons.js:69:8)
  at bound.module.exports.findOne (C:\Users\<user>\AppData\Roaming\npm\node_modules\sails\node_modules\waterline\lib\waterline\query\finder
s\basic.js:61:16)
  at bound [as findOne] (C:\Users\<user>\AppData\Roaming\npm\node_modules\sails\node_modules\lodash\dist\lodash.js:729:21)
  at [object Object].Deferred.exec (C:\Users\<user>\AppData\Roaming\npm\node_modules\sails\node_modules\waterline\lib\waterline\query\defer
red.js:493:16)
  at [object Object].promisified (<anonymous>:12:26)
  at [object Object].Deferred.toPromise (C:\Users\<user>\AppData\Roaming\npm\node_modules\sails\node_modules\waterline\lib\waterline\query\
deferred.js:502:61)
  at [object Object].Deferred.then (C:\Users\<user>\AppData\Roaming\npm\node_modules\sails\node_modules\waterline\lib\waterline\query\defer
red.js:513:15)
```

issue observed with sails (0.10.5) with  sails-mysql (0.10.6) ,

similar issues observed with sails-disk and sails-memory

```
error: A hook (`orm`) failed to load!
error: AdapterError: Connection is already registered
```

**Comment by IanVS** [comment]
I'm using [tape](https://github.com/substack/tape) for testing a sails-hook, and the fact that sails does not completely lower with a callback is causing problems.  In tape, tests will not complete if there are any async processes running, which seems to be different from Mocha (see: https://github.com/substack/tape/issues/164).  No matter how I try to `sails.lower`, the test stays hung.  I'm using `sails-memory`, if that makes a difference.

As a test, I tried:

```javascript
test('Teardown', function (t) {
  sails.lower(function() {
    console.log(sails);
  });
})
```

And the result was:

```
# Teardown

  |>   [a Sails app]
\___/  For help, see: http://links.sailsjs.org/docs

Tip: Use `sails.config` to access your app's runtime configuration.

1 Models:
User

1 Controllers:
UserController

14 Hooks:
moduleloader,logger,request,orm,blueprints,responses,controllers,sockets,policies,services,userconfig,http,userhooks,jsonapi
```

Which sure makes it look like there is still a live sails app, even inside the callback.  Any clues what's going on here?

**Comment by sailsbot** [comment]
Thanks for posting, @arkoak.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
